pkg/auth/providers/common/provider_util_test.go

```diff
@@ -41,3 +46,66 @@ func TestEscapeUUID(t *testing.T) {
 		})
 	}
 }
+
+func TestDecode(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name    string
+		input   any
+		output  any
+		wantErr bool
+	}{
+		{
+			name:    "successful decode",
+			input:   getMockAuthConfig(),
+			output:  &apimgmtv3.AuthConfig{},
+			wantErr: false,
+		},
+		{
+			name:    "unsuccessful decoder create",
+			input:   getMockAuthConfig(),
+			output:  apimgmtv3.AuthConfig{},
+			wantErr: true,
+		},
+		{
+			name:    "unsuccessful decode",
+			input:   "bogus input",
+			output:  &apimgmtv3.AuthConfig{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := common.Decode(tt.input, tt.output)
+			assert.Equal(t, err != nil, tt.wantErr)
+			if !tt.wantErr {
+				inputMap, _ := tt.input.(map[string]interface{})
```

Comment:
The tiniest nit (feel free to ignore): you have mixed usage of `map[string]any` and `map[string]interface{}`. Might want to standardize on the former, but it's immaterial otherwise.

Reply:
Yeah, if I update again (and tempt the ci gods), I can standardize.